### PR TITLE
[ace] Add patch to fix Visual Studio 2019 16.5 internal compiler error

### LINKED ISSUE #11430: FYI [angle] build failure with VS 2017 vc141 fatal error C1001: An internal error has occurred in the compiler #11387
Reference [angle] build failure with VS 2017 vc141 fatal error C1001: An internal error has occurred in the compiler #11387 
Reference [ace] vcpkg install ace:x64-windows fails on Visual Studio 2019 16.5  #10522 

Both are same issue, but different VS versions and toolset.

Reported VS 2017 issue: https://developercommunity.visualstudio.com/content/problem/1037098/vs-2017-fatal-error-c1001-an-internal-error-has-oc.html

Reported VS 2019 issue:
https://developercommunity.visualstudio.com/content/problem/954522/internal-compiler-error-1925286104.html

### PR DESCRIPTION
**Pull request by jwillemsen**
- What does your PR fix? Fixes https://github.com/microsoft/vcpkg/issues/11430

Fix internal compiler error by changing some ACE code

- Which triplets are supported/not supported? Have you updated the CI baseline?

Supported are now:

- - ace:arm64-windows
- - ace:x64-windows-static
- - ace:x64-windows
- - ace:x86-windows

### CODE REVIEW

File: ports/ace/portfile.cmake [hunk 1]
```diff
@@ -23,6 +23,11 @@ vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
     ARCHIVE ${ARCHIVE}
 )
+vcpkg_apply_patches(
+    SOURCE_PATH ${SOURCE_PATH}
+    PATCHES
+        "${CMAKE_CURRENT_LIST_DIR}/process_manager.patch"
```
Comment (NancyLi1013) [inline ports/ace/portfile.cmake:29]: The patch can be added to this function `vcpkg_extract_source_archive_ex()`.
```
vcpkg_extract_source_archive_ex(

...
PATCHES process_manager.patch
)
```
Could you please update this?

File: ports/ace/portfile.cmake [hunk 2]
```diff
@@ -23,6 +23,11 @@ vcpkg_extract_source_archive_ex(
...
+)
```
Comment (NancyLi1013) [inline ports/ace/portfile.cmake:31]: Could you please update this
```
  file(COPY ${ACE_ROOT}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/ace)
  file(RENAME ${CURRENT_PACKAGES_DIR}/share/ace/COPYING ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright)
```
as
`  file(INSTALL ${ACE_ROOT}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
`
?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/ace/portfile.cmake (hunk 2 above)
Comment (jwillemsen) [reply ports/ace/portfile.cmake:27]: Done, when you would use suggestion with your comment we could directly commit it on github

File: ports/ace/portfile.cmake (hunk 1 above)
Comment (jwillemsen) [reply ports/ace/portfile.cmake]: Done

File: ports/ace/portfile.cmake (hunk 2 above)
Comment (NancyLi1013) [reply ports/ace/portfile.cmake:27]: Thanks for your suggestions.

**Comment by jwillemsen** [comment]
Looks the osx error is something on azure, not related to my changes

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
@jwillemsen 
Thanks for your notice.
You can ignore the error. We will try to solve it.

**Comment by jwillemsen** [comment]
Any updates @NancyLi1013

**Review by NancyLi1013 — APPROVED** [review]

**Comment by jwillemsen** [comment]
Looks the arm64_windows failure is not related to my changes

**Comment by NancyLi1013** [comment]
@jwillemsen 
The failure on arm64-windows is not related with this PR. You can ignore it. We will resolve it later.

All features have passed with the following triplets:

- x86-windows
- x64-windows
- x64-widnows-static

**Comment by jwillemsen** [comment]
Any update on when this would get merged?

**Comment by NancyLi1013** [comment]
@strega-nil 
Could you please help review and merge this PR?

File: ports/ace/portfile.cmake [hunk 3]
```diff
@@ -22,6 +22,7 @@ endif()
 vcpkg_extract_source_archive_ex(
     OUT_SOURCE_PATH SOURCE_PATH
     ARCHIVE ${ARCHIVE}
+    PATCHES "${CMAKE_CURRENT_LIST_DIR}/process_manager.patch"
```
Comment (strega-nil) [inline ports/ace/portfile.cmake:25]:
```suggestion
    PATCHES
        "${CMAKE_CURRENT_LIST_DIR}/process_manager.patch" # Fix MSVC 16.5 ICE
```

**Comment by jwillemsen** [comment]
Is there anything I could do to get this merged?

**Comment by NancyLi1013** [comment]
@strega-nil 
Could you please help review if there is anything needed to be done for this PR?

**Comment by dan-shaw** [comment]
LGTM

**Pull request merged**